docs/api/python.rst

```diff
@@ -0,0 +1,66 @@
+Python APIs
+===========
+
+``Lance`` is a columnar format that is specifically designed for efficient
```

Comment:
Since we're talking about datasets it might be more precise to refer to it as a table format here rather than a columnar format.